pandas/core/arrays/integer.py

```diff
@@ -644,9 +644,20 @@ def integer_arithmetic_method(self, other):
         name = '__{name}__'.format(name=op.__name__)
         return set_function_name(integer_arithmetic_method, name, cls)
 
+    @classmethod
+    def _create_unary_method(cls, op):
+        def integer_unary_method(self):
+            with np.errstate(all='ignore'):
+                result = op(self._data)
+            return type(self)(result, self._mask)
```

Comment:
should this use _from_sequence?

Reply:
`from_sequence` can't accept mask. Other methods (e.g. `__getitem__`) follow the same manner to make new array using the current mask.